WL-3017 Removed hostname(server-url) from site-info urls while storing in the DB.

ouit0408 opened the pull request:
And appended the server url in the site-info url on the edit screen for
home page.

buckett commented on web-portlet/src/java/org/sakaiproject/portlets/PortletIFrame.java:
```diff
@@ -636,7 +637,14 @@ else if (SPECIAL_WORKSITE.equals(special))
+                                                   String serverUrl = ServerConfigurationService.getServerUrl();
+                                                   String serverName = new URL(serverUrl).getHost();
+                                                   Pattern checkPattern = Pattern.compile(String.format("^(https?:)?//%s:?\\d*/", serverName));
+                                                   if(!checkPattern.matcher(infoUrl).find()){
```
If I have an info URL of ftp://myhost.ox.ac.uk/timetable then you will end up prepending the `serverUrl` onto it.

ouit0408 replied:
assuming we are filtering 'ftp' url as well before storing in the database , for this method it should ideally be absolute path always.

buckett replied:
But we already have existing values in the database these might not have been filtered.

ouit0408 replied:
so if there is infoUrl with 'ftp', and serverUrl has 'http' and if we filter 'ftp' , will it be correct to append the serverUrl with http to it?

ouit0408 replied:
so for urls having ftp in their infoUrl what will be the flow like? should we strip ftp and save back the absolute path in the db and how it should be displayed on the edit screen?

buckett commented on web-portlet/src/java/org/sakaiproject/portlets/PortletIFrame.java:
```diff
@@ -904,7 +912,18 @@ else if (newPageTitle.length() > MAX_TITLE_LENGTH)
                 if ((infoUrl != null) && (infoUrl.length() > 0) && (!infoUrl.startsWith("/")) && (infoUrl.indexOf("://") == -1))
                 {
-                    infoUrl = "http://" + infoUrl;
+                    infoUrl = "//" + infoUrl;
```
Although this seems good, I'm not 100% sure it's the right thing todo. I wonder if this was there so that if someone enters a URL of www.university.edu/course/STAT-101 it prepends the http:// onto it. However if the URL is for a different server there's no guarantee the server's protocol is any better than http.

What do you think?

ouit0408 replied:
there is a 'validateUrl' method called when infourl is edited which generates error for this www.university.edu/course/STAT-101 url but if 'http://www.university.edu/course/STAT-101' this will get added. So we should not let user add such url?

ouit0408 committed:
WL-3017 Removed hostname from site-info urls while storing in the DB.

And appended the server url in the site-info url on the edit screen for
home page.
Implemented the review comments.

ouit0408 commented:
Implemented comments and have pushed the changes.

buckett commented on web-portlet/src/java/org/sakaiproject/portlets/PortletIFrame.java:
```diff
@@ -636,7 +637,12 @@ else if (SPECIAL_WORKSITE.equals(special))
-						    context.put("info_url", FormattedText.escapeHtmlFormattedTextarea(infoUrl));
+                                                   //Check if infoUrl is relative? and prepend the server url
+                                                   String serverUrl = ServerConfigurationService.getServerUrl();
+                                                   if(infoUrl.startsWith("/") && infoUrl.indexOf("://") == -1){
```
This fails with about:blank as the URL.

ouit0408 replied:
I will not add server url to 'about:blank' url as this does not start with /
